homeassistant/components/subaru/sensor.py

```diff
@@ -271,14 +266,9 @@ def available(self) -> bool:
             return False
         return last_update_success
 
-    def get_current_value(self):
+    def get_current_value(self) -> None | int | float:
         """Get raw value from the coordinator."""
-        value = self.coordinator.data[self.vin][VEHICLE_STATUS].get(self.data_field)
-        if value in sc.BAD_SENSOR_VALUES:
-            value = None
-        if isinstance(value, str):
-            if "." in value:
-                value = float(value)
-            else:
-                value = int(value)
+        value = self.coordinator.data[self.vin]
+        value = value[VEHICLE_STATUS].get(self.entity_description.key)
+        _LOGGER.debug("Raw value for %s: %s", self.name, value)
```

Comment:
We don't want to do side effects like logging in entity state properties.

Please remove the log or move it to a state update callback.